Dockerfile.dapper

```diff
@@ -1,4 +1,4 @@
-ARG GOLANG=golang:1.23.6-alpine3.21
+ARG GOLANG=golang:1.23.8-alpine3.20
```

Comment:
Do no change the golang version like this. You must use the version of golang used by upstream:
https://github.com/kubernetes/kubernetes/blob/v1.32.3/build/dependencies.yaml#L117-L119

Reply:
I noticed the CI fail because of that, but as I said in https://github.com/k3s-io/k3s/pull/12126#issuecomment-2798803666, this fails if we don't updated go

```
$ go build
go: github.com/flannel-io/flannel@v0.26.6 requires go >= 1.23.7 (running go 1.23.6; GOTOOLCHAIN=local)
``` 

It's probably easiest for y'all to close this and handle it on your end, since I can't cut a new flannel release with a relaxed version requirement